src/core/observer/watcher.js

```diff
@@ -81,7 +82,7 @@ export default class Watcher {
     } else {
       this.getter = parsePath(expOrFn)
       if (!this.getter) {
-        this.getter = function () {}
+        this.getter = noop
```

Comment:
This one should stay as `function () {}` because `noop` is an identity function. Look at line 109: `value = this.getter.call(vm, vm)` would resolve to `vm`

Reply:
I am sorry I can't understand why `noop` is an indetity function, and why noop different from `function () {}`
```
export function noop (a?: any, b?: any, c?: any) {}
```
Thanks for your time!

Reply:
omg, you are right, I misread in the code there is another function called noop in write.js but that one is not exported!

Reply:
Maybe should rename `noop` to `indentity` in write.js like `shared/util`, `noop` is misleading.

Reply:
Identity function should return its first argument.